include/CMakeLists.txt

```diff
@@ -51,8 +51,13 @@ target_sources(cpr PRIVATE
 )
 
 # Filesystem
-if ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1) OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+include(CheckIncludeFileCXX)
```

Comment:
Maybe we can replace this with a `find_package` call for [boost](https://cmake.org/cmake/help/latest/module/FindBoost.html).

Reply:
Sorry for delay, I’ve been very sick these days and just got back to work.

So now CMake uses FindBoost first to find Boost.Filesystem in case we set CPR_USE_BOOST_FILESYSTEM option to ON (OFF by default) and if this option is disabled - we try to add Standard.Library instead.

Added CI run with Boost for "autossl" build configuration. Not sure we will need the full test matrix with other configurations/platforms for this Boost option yet.